`getEgoVehicleId` does not compile in Linux

It looks like there's an issue with the definition of the ego vehicle name grabber function introduced in PR #100

```
[ 10%] Building CXX object monodrive/core/CMakeFiles/monodrive.dir/src/Simulator.cpp.o
/home/bradnull/projects/monodrive-client/monodrive/core/src/Simulator.cpp: In member function ‘std::__cxx11::string Simulator::getEgoVehicleId()’:
/home/bradnull/projects/monodrive-client/monodrive/core/src/Simulator.cpp:385:2: error: cannot bind non-const lvalue reference of type ‘Simulator::getEgoVehicleId()::<lambda(nlohmann::json&)>&’ to an rvalue of type ‘Simulator::getEgoVehicleId()::<lambda(nlohmann::json&)>’
  };
  ^
```

--- a/monodrive/core/src/Simulator.cpp
+++ b/monodrive/core/src/Simulator.cpp
@@ -366,7 +366,7 @@ bool Simulator::sendControl(float forward, float right, float brake, int mode)
 }
 
 std::string Simulator::getEgoVehicleId() {
-	auto& getEgoVehicleName = [](nlohmann::json& vehicles) {
+	const auto getEgoVehicleName = [](nlohmann::json& vehicles) {
 		for (auto& obj : vehicles) {
 			nlohmann::json vehicle = obj;
 			if (obj.find("state") != obj.end()) {

